ios/Classes/SwiftMparticleFlutterSdkPlugin.swift

```diff
@@ -244,6 +244,121 @@ public class SwiftMparticleFlutterSdkPlugin: NSObject, FlutterPlugin {
+                        promotion.position = rawPromotion["position"] as? String
+
+                        container .addPromotion(promotion)
+                    }
+                }
+
+                event = MPCommerceEvent.init(promotionContainer: container)
+            } else {
+                event = MPCommerceEvent.init(impressionName: nil, product: nil)
+            }
+            // Optional Products on Commerce Event
+            if let rawProducts = callArguments["products"] as? [[String: Any]] {
+                for rawProduct in rawProducts {
+                    if let name = rawProduct["name"] as? String,
+                       let sku = rawProduct["sku"] as? String,
+                       let quantity = rawProduct["quantity"] as? NSNumber,
+                       let price = rawProduct["price"] as? NSNumber {
+                        let newProduct = MPProduct.init(name: name,
+                                                        sku: sku,
+                                                        quantity: quantity,
+                                                        price: price)
+                        event.addProduct(newProduct)
+                    }
+                }
+            }
+
+            // Optional Transaction Attributes
+            if let rawTransactionAttributes = callArguments["transactionAttributes"] as? [String: Any] {
+                let attributes = MPTransactionAttributes.init()
+                if let affiliation = rawTransactionAttributes["affiliation"] as? String {
+                    attributes.affiliation = affiliation
+                }
+                if let couponCode = rawTransactionAttributes["couponCode"] as? String {
+                    attributes.couponCode = couponCode
+                }
+                if let shipping = rawTransactionAttributes["shipping"] as? NSNumber {
+                    attributes.shipping = shipping
+                }
+                if let tax = rawTransactionAttributes["tax"] as? NSNumber {
+                    attributes.tax = tax
+                }
+                if let revenue = rawTransactionAttributes["revenue"] as? NSNumber {
+                    attributes.revenue = revenue
+                }
+                if let transactionId = rawTransactionAttributes["transactionId"] as? String {
+                    attributes.transactionId = transactionId
+                }
+
+                event.transactionAttributes = attributes
+            }
+
+            if let rawImpressions = callArguments["impressions"] as? [[String: Any]] {
+                for rawImpression in rawImpressions {
+                    if let listName = rawImpression["impressionListName"] as? String,
+                       let rawProducts = rawImpression["products"] as? [[String: Any]] {
+                        for rawProduct in rawProducts {
+                            if let name = rawProduct["name"] as? String,
```

Comment:
nit: it'd probably be cleaner if you could abstract this MPProduct deserialization into a method or something so we can reuse it since we have the same logic earlier in the method